singletons/utils.js

```diff
@@ -156,19 +156,75 @@ module.exports = (function (Module) {
 		 * Returns a formatted string, specifying an amount of time delta from current date to provided date.
 		 * @param {sb.Date|Date|number} target
 		 * @param {boolean} [skipAffixes] if true, the affixes "in X hours" or "X hours ago" will be omitted
+		 * @param {sb.Date} [deltaTo] If set, calculate time delta between target and deltaTo. If undefined, calculate
+		 * delta between target and the current time.
 		 * @returns {string}
 		 */
-		timeDelta (target, skipAffixes = false) {
-			const now = new sb.Date();
-			if (sb.Date.equals(now, target)) {
+		timeDelta (target, skipAffixes = false, deltaTo = undefined) {
+			if (deltaTo === undefined) {
+				deltaTo = new sb.Date();
+			}
+
+			if (sb.Date.equals(deltaTo, target)) {
 				return "right now!";
 			}
 
+			if (typeof target === "number") {
+				target = new sb.Date(target);
+			}
+			else {
+				target = new sb.Date(target.valueOf());
+			}
+
+			let prefix;
+			let suffix;
+			let earlier;
+			let later;
+			if (target > deltaTo) {
+				prefix = "in ";
+				suffix = "";
+				earlier = deltaTo;
+				later = target;
+			} else {
+				prefix = "";
+				suffix = " ago";
+				earlier = target;
+				later = deltaTo;
+			}
+
 			let string;
-			const delta = Math.abs(now.valueOf() - target.valueOf());
-			const [prefix, suffix] = (target > now) ? ["in ", ""] : ["", " ago"];
+			const delta = later.valueOf() - earlier.valueOf();
+
+			// special case for >1 year that respects leap years.
+			const earlierPlusOneYear = earlier.clone();
+			earlierPlusOneYear.setUTCFullYear(earlierPlusOneYear.getUTCFullYear() + 1);
+
+			// this rounds the `later` date to a value so that the difference between `earlier` and `laterRounded` is
+			// a whole number of seconds. Avoids issues with floating point rounding stuff where the difference
+			// between the dates is just below a whole number of years.
+			const roundedDelta = this.round(delta, -4);
+			const laterRounded = later.clone();
+			laterRounded.setTime(earlier.valueOf() + roundedDelta);
 
-			if (delta < Utils.timeUnits.s.ms) {
+			if (earlierPlusOneYear <= laterRounded) {
```

Comment:
I feel like this branch of logic could have been incorporated into the original if-branch for when the difference is at least 1 year - this way we don't have to "pollute" the function scope with variables like `earlier` and `later` and all the calculations done outside of the branch.